src/engine/Settings.cs

```diff
@@ -1,15 +1,26 @@
-﻿using Godot;
+﻿using System;
+using Godot;
 using Newtonsoft.Json;
 
 /// <summary>
-///   Main object for containing player changeable game settings
+///   Singleton class that handles storing and applying player changeable game settings.
```

Comment:
I didn't originally say that this is a singleton because the design is simpler when this isn't really a singleton.
So all of the duplicate fields etc. could be avoided by having Settings be IClonable.

Reply:
Technically it had a private constructor before, so for all intents and purposes it basically is a singleton, since you can't actually clone the class as far as I know. I'm not that familiar with C# though so maybe I'm interpreting that wrong.

Reply:
Yes, that was done to protect things. However I never thought that things like adding a clone method for Settings was out of the picture, so I didn't refer to Settings as a singleton anywhere to avoid the confusion because it actually isn't (in the reset settings code there is a fresh instance created to apply the defaults again).